### ISSUE TITLE
`byte_char_slices` should trigger on arrays

### ISSUE THREAD

**Issue by poseidon-rises** [rust-lang/rust-clippy#16759]
### Summary

I know that it's called `slices`, but I believe that the `byte_char_slices` should also trigger on sized arrays.

### Lint Name

byte_char_slices

### Reproducer

If I created this code here:

```rust
// Some function that validates a string represented as 3 u32s
fn check_str(bytes: [u32; 3]) -> bool {
     // Some other garbage
    const START: u32 = u32::from_le_bytes([b'W', b'O', b'R', b'K']);
    if bytes.starts_with(START) { /* */ }
    // Some more garbage
}
```
I would the lint to trigger, suggesting a replacement of `*b"WORK"`. Instead, no lint is triggered.

### Version

```text
rustc 1.96.0-nightly (91021ccc7 2026-03-17)
binary: rustc
commit-hash: 91021ccc790478a1a89c003e7d32b8d155ae6aae
commit-date: 2026-03-17
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.0
```
